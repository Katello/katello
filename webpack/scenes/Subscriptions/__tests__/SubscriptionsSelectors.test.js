import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';
import {
  selectSubscriptionsState,
  selectSearchQuery,
  selectDeleteModalOpened,
  selectDeleteButtonDisabled,
  selectSubscriptionsTasks,
  selectTaskModalOpened,
} from '../SubscriptionsSelectors';

const state = {
  katello: {
    subscriptions: {
      searchQuery: 'some-query',
      deleteModalOpened: false,
      taskModalOpened: false,
      deleteButtonDisabled: true,
      tasks: ['task1', 'task2'],
    },
  },
};

const fixtures = {
  'should select the subscriptions state': () => selectSubscriptionsState(state),
  'should select search-query': () => selectSearchQuery(state),
  'should select delete-modal-opened': () => selectDeleteModalOpened(state),
  'should select task-modal-opened': () => selectTaskModalOpened(state),
  'should select delete-button-disabled': () => selectDeleteButtonDisabled(state),
  'should select subscriptions tasks': () => selectSubscriptionsTasks(state),
};

describe('Subscriptions selectors', () => testSelectorsSnapshotWithFixtures(fixtures));
