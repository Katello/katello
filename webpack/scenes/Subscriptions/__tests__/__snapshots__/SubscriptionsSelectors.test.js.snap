// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriptions selectors should select delete-button-disabled 1`] = `true`;

exports[`Subscriptions selectors should select delete-modal-opened 1`] = `false`;

exports[`Subscriptions selectors should select search-query 1`] = `"some-query"`;

exports[`Subscriptions selectors should select subscriptions tasks 1`] = `
Array [
  "task1",
  "task2",
]
`;

exports[`Subscriptions selectors should select task-modal-opened 1`] = `false`;

exports[`Subscriptions selectors should select the subscriptions state 1`] = `
Object {
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "searchQuery": "some-query",
  "taskModalOpened": false,
  "tasks": Array [
    "task1",
    "task2",
  ],
}
`;
